app/services/submit_vehicle_service.rb

```diff
@@ -0,0 +1,39 @@
+class SubmitVehicleService < BaseCfeService
+  def self.call(cfe_estimate_id, cfe_session_data)
+    new.call(cfe_estimate_id, cfe_session_data)
+  end
+
+  def call(cfe_estimate_id, cfe_session_data)
+    finance_form = Flow::Vehicle::FinanceHandler.model(cfe_session_data)
+
+    finance_form.vehicle_pcp.present? ? submit_vehicle_finance_data(cfe_estimate_id, cfe_session_data) : submit_vehicle_value_data(cfe_estimate_id, cfe_session_data)
+  end
+
+  def submit_vehicle_value_data(cfe_estimate_id, cfe_session_data)
+    value_form = Flow::Vehicle::ValueHandler.model(cfe_session_data)
+    return if value_form.vehicle_value.blank?
+
+    cfe_connection.create_vehicle cfe_estimate_id, date_of_purchase: Time.zone.today.to_date,
+                                                   value: value_form.vehicle_value,
+                                                   loan_amount_outstanding: 0,
+                                                   in_regular_use: value_form.vehicle_in_regular_use
+  end
+
+  def submit_vehicle_finance_data(cfe_estimate_id, cfe_session_data)
+    finance_form = Flow::Vehicle::FinanceHandler.model(cfe_session_data)
+
+    # line below is missing branch coverage. However I am not sure it is required. we validate on the page
+    # that a value is entered if vehicle_pcp is selected, if no value is entered the estimate cannot progress and therefore no submission
+    # to CFE is possible.
```

Comment:
Again, if we've got logic that's not being tested and _can't_ be tested by a high-level spec, we can either knock up a quick unit test, or, if you think this logic is completely redundant, you could just remove the untested logic.

Reply:
removed after ensuring that it didnt break anything else and that I was sure of my logic